src/WebJobs.Extensions.DurableTask/ManagedIdentityTokenSource.cs

```diff
@@ -34,13 +36,40 @@ public ManagedIdentityTokenSource(string resource)
         [JsonProperty("resource")]
         public string Resource { get; }
 
+        /// <summary>
+        /// The azure credential options that a user can configure when authenticating.
+        /// </summary>
+        [JsonProperty("options")]
+        public ManagedIdentityOptions Options { get; }
+
         /// <inheritdoc/>
         public async Task<string> GetTokenAsync()
         {
             var scopes = new string[] { this.Resource };
             TokenRequestContext context = new TokenRequestContext(scopes);
 
-            DefaultAzureCredential defaultCredential = new DefaultAzureCredential();
+            DefaultAzureCredential defaultCredential;
+            DefaultAzureCredentialOptions defaultAzureCredentialOptions = new DefaultAzureCredentialOptions();
+
+            if (this.Options != null && ((this.Options.AuthorityHost != null) || (!string.IsNullOrEmpty(this.Options.TenantId))))
+            {
+                if (this.Options.AuthorityHost != null)
```

Comment:
It seems like the body of this if statement could just be the code used, and we don't need the surrounding if-else statement? Essentially we use the default credential options and only edit them if the fields on this.Options are populated?

The only difference would be needing to stick `?.` operators after `this.Options`.